client/src/main/scala/client/BackPressuredWebSocketClient.scala

```diff
@@ -1,58 +1,67 @@
 package client
 
-import monifu.concurrent.Scheduler
-import monifu.reactive.{Ack, Observable, Observer, Subscriber}
+import monix.execution.cancelables.BooleanCancelable
+import monix.execution.{Ack, Cancelable, Scheduler}
+import monix.reactive.observers.Subscriber
+import monix.reactive.{Observable, Observer}
 import org.reactivestreams.Subscription
 import org.scalajs.dom.raw.MessageEvent
 import org.scalajs.dom.{CloseEvent, ErrorEvent, Event, WebSocket}
+
 import scala.concurrent._
-import concurrent.duration._
+import scala.concurrent.duration._
 
 
-final class BackPressuredWebSocketClient private (url: String)
-  extends Observable[String] { self =>
+final class BackPressuredWebSocketClient private(url: String)
+  extends Observable[String] {
+  self =>
 
   private def createChannel(webSocket: WebSocket) =
-    Observable.create[String] { subscriber =>
-      import subscriber.scheduler
-      val downstream = Observer.toReactiveSubscriber(subscriber)
-
-      try {
-        webSocket.onopen = (event: Event) => {
-          downstream.onSubscribe(new Subscription {
-            def cancel(): Unit =
-              closeConnection(webSocket)
-
-            def request(n: Long): Unit = {
-              webSocket.send(n.toString)
-            }
-          })
-        }
-
-        webSocket.onerror = (event: ErrorEvent) => {
-          downstream.onError(BackPressuredWebSocketClient.Exception(event.message))
-        }
-
-        webSocket.onclose = (event: CloseEvent) => {
-          downstream.onComplete()
+    new Observable[String] {
+      override def unsafeSubscribeFn(subscriber: Subscriber[String]): Cancelable = {
+        import subscriber.scheduler
+        val downstream = Observer.toReactiveSubscriber(subscriber)
+
+        try {
+          webSocket.onopen = (event: Event) => {
+            downstream.onSubscribe(new Subscription {
+              def cancel(): Unit =
+                closeConnection(webSocket)
+
+              def request(n: Long): Unit = {
+                webSocket.send(n.toString)
+              }
+            })
+          }
+
+          webSocket.onerror = (event: ErrorEvent) => {
+            downstream.onError(BackPressuredWebSocketClient.Exception(event.message))
+          }
+
+          webSocket.onclose = (event: CloseEvent) => {
+            downstream.onComplete()
+          }
+
+          webSocket.onmessage = (event: MessageEvent) => {
+            downstream.onNext(event.data.asInstanceOf[String])
+          }
+        } catch {
+          case ex: Throwable =>
+            downstream.onError(ex)
         }
 
-        webSocket.onmessage = (event: MessageEvent) => {
-          downstream.onNext(event.data.asInstanceOf[String])
-        }
-      }
-      catch {
-        case ex: Throwable =>
-          downstream.onError(ex)
+        BooleanCancelable()
```

Comment:
This shouldn't be an empty cancelable.

Amongst the biggest changes in Monix 2.0 is the behavior on cancelation. As you have noticed, `unsafeSubscribeFn` now returns a `Cancelable`. The purpose of this `Cancelable` is to cancel a subscription, whereas in Monifu 1.x we were doing this by a trick, by calling `takeWhileNotCanceled(cancelable)`, which requires the producer to push another item into the consumer in order to notice that it has been canceled. Therefore in Monifu 1.x the cancelation was NOT immediate.

This is why in `monifu-sample` we've got a `def createChannel(webSocket: WebSocket)` that takes an already initialized `WebSocket`. Because that `WebSocket` reference we need to close whenever something happens (like a `timeout` being triggered). If you'll see below, we have to detect problems and take care of calling `closeConnection(webSocket)` ourselves. But now that we return a `Cancelable`, this `Cancelable` can take care of calling `closeConnection(webSocket)`. In other words:

1. Make this returned cancelable call the logic in `closeConnection(webSocket)` (and note that it can be a plain `Cancelable` and not a `BooleanCancelable`)
2. Whatever subscriber is being subscribed to this `Observable`, we need to wrap it into something that also calls `closeConnection` if `onError` or `onComplete` happens (we should no longer rely on `doOnCancel`, because its meaning has changed btw ;))
3. The `createChannel()` function should no longer receive that `WebSocket` reference. All logic should be encapsulated in this function. The `closeConnection` function itself should be an inner function of this function.